RetailCoder.VBE/Refactorings/Rename/RenameModel.cs

```diff
@@ -76,5 +108,41 @@ public void PromptIfTargetImplementsInterface(ref Declaration target)
 
             target = interfaceMember;
         }
+
+        internal class Rewriters
+        {
+            Dictionary<string, IModuleRewriter> _rewriters;
+            RubberduckParserState _state;
+
+            public Rewriters(RubberduckParserState state)
+            {
+                _rewriters = new Dictionary<string, IModuleRewriter>();
+                _state = state;
+            }
+
+            public IModuleRewriter GetRewriter(QualifiedModuleName qmn)
+            {
+                IModuleRewriter rewriter;
+                if (_rewriters.ContainsKey(qmn.Name))
+                {
+                    _rewriters.TryGetValue(qmn.Name, out rewriter);
+                }
+                else
+                {
+                    rewriter = _state.GetRewriter(qmn);
+                    _rewriters.Add(qmn.Name, rewriter);
+                }
+                return rewriter;
+            }
+
+            public void Rewrite()
+            {
+                foreach(var rewriter in _rewriters.Values)
```

Comment:
this could be dangerous, because we might need to apply rewriters in a specific order